### ISSUE TITLE
Use of priors with ITS data

### ISSUE THREAD

**Issue by ramiroricardo** [benjjneb/dada2#1599]
Dear all,

we have an ITS dataset to analyse and there is a species of interest that we would like to be more sensitive in detecting. We can go into pooling, but we were also wondering about the use of priors and if it makes sense in ITS data.

In our case, we are analyzing only forward reads and using `truncQ` and `maxEE`, but not `truncLen`. This will lead to reads of variable length, in principle even for the exact same taxa. In #496, it is indicated that priors have to have the exact same length as the processed reads, which is fine for 16S data, but I am guessing won't work for ITS data. This would mean, we should not (yet) use priors with ITS data. 

Am I missing something / is there a way in which priors can still be included when analyzing ITS data?

Thanks

**Comment by benjjneb** [comment]
My first suggestions is to stop using `truncQ` to get rid of the artificial length variation that filtering technique introduces. It is almost never the right choice when analyzing data with DADA2.

If that is eliminated, then it becomes possible to use priors with ITS data, although one does have to be precise about setting the prior sequences to be of the same length as they will be in the data. However, if the target species ITS sequence is known, that should be achievable. Note that if working with paired-end data, you'll need to create priors separately for the forward and reverse reads.

**Comment by ramiroricardo** [comment]
Thanks for your quick reply. Just a further question: we were using `truncQ` as it is recommended in the DADA2 ITS workflow. Thus why would it not be recommended? because we are using forward reads only?

**Comment by benjjneb** [comment]
Hm, I see that there now. By default `truncQ=2` is enforced anyway by `filterAndTrim`, as several variations of Cassava/Illumina would start assigning Q=2 when it no longer had any idea what the bases were. If you have just been using `truncQ=2`, I'm not too concerned actually as that should introduce relatively little length variation. But the way the tutorial is written is looks like `truncQ=XX` is a parameter we recommended potentially tuning, and that's not the case even for ITS data.

**Comment by ramiroricardo** [comment]
Thanks for the reply. Indeed we started wondering about changing `truncQ` after seeing this paper, where several values are tested:
https://insight.jci.org/articles/view/151663

**Comment by markschl** [comment]
> If you have just been using `truncQ=2`, I'm not too concerned actually as that should introduce relatively little length variation. But the way the tutorial is written is looks like `truncQ=XX` is a parameter we recommended potentially tuning, and that's not the case even for ITS data.

Is it possible to say which amount of length variation of reads (due to truncQ) is still acceptable? I'm not sure I have a full understanding of all the problems that could arise in different steps. But maybe I missed some important documentation somewhere. This question is not related to priors, although I have been using pseudo-pooling where they also play a role.

**Comment by benjjneb** [comment]
@markschl The fraction of reads affected by `truncQ` -- call it Q -- will increase the limit of detection threshold of rare variants (the number of reads required for detection) by about the factor 1/Q.

**Comment by benjjneb** [comment]
@markschl Yes, in the case of phased primers (also known as heterogeneity spacers), you need to use cutadapt or some other external tool to remove the primers, and it is then ideal to truncate to a fixed length after removing the primers to get rid of the spurious length variation the different length primers introduce into the trimmed reads.

The issue is that the high length-variability of ITS makes this is hard to do. If the truncation length is chosen longer than the shortest real amplicons in the data, then the length variability of those amplicons won't be fixed. But if set to the minimum length of the ITS region being targeted, this might end up truncating the bulk of the sequences to <1/3rd their length, majorly affecting resolution.

I guess the solution might be that the short sequences read into the reverse primer, and so trimming both the forward and reverse primers will work for the shorter (< read length) amplicons, while the longer (> read length) amplicons could then all be truncated to 297-Forward_primer_length.
